src/main/java/org/cryptimeleon/craco/sig/sps/agho11/SPSAGHO11SignatureScheme.java

```diff
@@ -45,13 +45,19 @@ public SPSAGHO11SignatureScheme(Representation repr) {
         new ReprUtil(this).deserialize(repr);
     }
 
+
     @Override
     public SignatureKeyPair<SPSAGHO11VerificationKey, SPSAGHO11SigningKey> generateKeyPair(int numberOfMessages)
     {
         return generateKeyPair(numberOfMessages, 2);
     }
 
-    @Override
+    /**
+     * Generates a key pair for signing n blocks of messages with {@code  messageBlockLengths}
+     * with each signature.
+     *
+     * @param messageBlockLengths the length of the individual MessageBlocks this scheme accepts as input.
+     */
     public SignatureKeyPair<SPSAGHO11VerificationKey, SPSAGHO11SigningKey> generateKeyPair(int... messageBlockLengths) {
```

Comment:
could you please be more precise in the exception "The given messageBlockLengths do not match the public parameters".

Reply:
Give the position that fails for example

Reply:
i have expanded the message checks and exception messages now.